### ISSUE TITLE
TypeSpecServerNotFoundException doesn't provide enough troubleshooting information

### ISSUE THREAD

**Issue by abatishchev** [microsoft/typespec#1904]
When VS extension can't bind the lang server and throws TypeSpecServerNotFoundException, it looks like this in VS logs:

```
Microsoft.TypeSpec.VisualStudio.TypeSpecServerNotFoundException: TypeSpec server executable was not found: 'tsp-server.cmd' is not found. Make sure either:
- typespec is installed locally at the root of this workspace or in a parent directory.
- typespec is installed globally with `npm install -g @typespec/compiler'.
- typespec server path is configured with https://github.com/microsoft/typespec/blob/main/packages/typespec-vs/README.md#configure-typespec-visual-studio-extension. ---> System.ComponentModel.Win32Exception: The system cannot find the file specified
   at System.Diagnostics.Process.StartWithCreateProcess(ProcessStartInfo startInfo)
   at System.Diagnostics.Process.Start(ProcessStartInfo startInfo)
   at Microsoft.TypeSpec.VisualStudio.LanguageClient.<ActivateAsync>d__23.MoveNext() in D:\a\_work\1\s\packages\typespec-vs\src\VSExtension.cs:line 98
 --- End of inner exception stack trace ---
   at Microsoft.TypeSpec.VisualStudio.LanguageClient.<ActivateAsync>d__23.MoveNext() in D:\a\_work\1\s\packages\typespec-vs\src\VSExtension.cs:line 108
--- End of stack trace from previous location where exception was thrown ---
```

All it says that it was looking for `tsp-server.cmd` which it couldn't find. But doesn't say where it was looking for and what should be done to make it working. In my case the right fix is this:

```json
{
  "typespec.tsp-server.path": "D:\\Projects\\ATS-RP\\src\\TypeSpec\\typespec\\node_modules\\@typespec\\compiler"
}
```
File: D:\Projects\ATS-RP\.vs\VSWorkspaceSettings.json.

**Comment by abatishchev** [comment]
I tried to address the issue in https://github.com/microsoft/typespec/pull/1902 but after talking to @nguerrera we agreed it might not be the right course of action.

**Comment by markcowl** [comment]
@abatishchev The message contaisn this link: https://github.com/microsoft/typespec/blob/main/packages/typespec-vs/README.md#configure-typespec-visual-studio-extension Do you feel like there isn't anough information here, or that the information from this link should be included in the message?  We feel like the message is already long and this would make it too long, but if there's a wording change you feel would be more clear, can you specify it?

**Comment by abatishchev** [comment]
Hi Mark.
You're right on both counts: the error message is too long, however it does provide enough troubleshooting information, at least the last link would point me to the doc which would help to address the issue I was facing.

---

To answer your question, what would help me is enriching the first part of the exception message. Not just:

> TypeSpec server executable was not found: 'tsp-server.cmd' is not found.

But something like:

> TypeSpec server executable was not found: 'tsp-server.cmd' is not found in path 'xyz'.

I've changed the exception to do just that in https://github.com/microsoft/typespec/pull/1902 but Nick didn't like it too much, asked to file an issue instead.

**To summarize:** currently the exception message doesn't provide enough information about "what's wrong" but provides enough information "how to fix that". I hope you see what I mean.
